contracts/oasisContracts/KyberOasisReserve.sol

```diff
@@ -0,0 +1,252 @@
+
+        return true;
+    }
+
+    event TradeEnabled(bool enable);
+
+    function enableTrade() public onlyAdmin returns(bool) {
+        tradeEnabled = true;
+        TradeEnabled(true);
+
+        return true;
+    }
+
+    function disableTrade() public onlyAlerter returns(bool) {
+        tradeEnabled = false;
+        TradeEnabled(false);
+
+        return true;
+    }
+
+    event KyberNetworkSet(address kyberNetwork);
+
+    function setKyberNetwork(address _kyberNetwork) public onlyAdmin {
+        require(_kyberNetwork != address(0));
+
+        kyberNetwork = _kyberNetwork;
+        KyberNetworkSet(kyberNetwork);
+    }
+
+    event OtcSet(address otc);
+
+    function setOtc(OtcInterface _otc) public onlyAdmin {
+        require(_otc != address(0));
+
+        wethToken.approve(otc, 0);
+        tradeToken.approve(otc, 0);
+        wethToken.approve(_otc, 2**255);
+        tradeToken.approve(_otc, 2**255);
+
+        otc = _otc;
+        OtcSet(otc);
+    }
+
+    event FeeBpsSet(uint feeBps);
+
+    function setFeeBps(uint _feeBps) public onlyAdmin {
+        require(_feeBps < 10000);
+
+        feeBps = _feeBps;
+        FeeBpsSet(feeBps);
+    }
+
+    function valueAfterReducingFee(uint val) public view returns(uint) {
+        require(val <= MAX_QTY);
+        return ((10000 - feeBps) * val) / 10000;
+    }
+
+    function valueBeforeFeesWereReduced(uint val) public view returns(uint) {
+        require(val <= MAX_QTY);
+        return val * 10000 / (10000 - feeBps);
+    }
+
+    function getConversionRate(ERC20 src, ERC20 dest, uint srcQty, uint blockNumber) public view returns(uint) {
+        uint  rate;
+        uint  actualSrcQty;
+        ERC20 wrappedSrc;
+        ERC20 wrappedDest;
+        uint  bestOfferId;
+        uint  bestOfferSrcQty;
+        uint  bestOfferDestQty;
+
+        blockNumber;
+
+        if (!tradeEnabled) return 0;
+        if ((tradeToken != src) && (tradeToken != dest)) return 0;
+
+        if (src == ETH_TOKEN_ADDRESS) {
+            wrappedSrc = wethToken;
+            wrappedDest = dest;
+            actualSrcQty = srcQty;
+        } else if (dest == ETH_TOKEN_ADDRESS) {
+            wrappedSrc = src;
+            wrappedDest = wethToken;
+
+            if (srcQty < MIN_TRADE_TOKEN_SRC_AMOUNT) {
+                /* Assuming token is stable, use a minimal amount to get rate also for small token quant. */
+                actualSrcQty = MIN_TRADE_TOKEN_SRC_AMOUNT;
+            } else {
+                actualSrcQty = srcQty;
+            }
+        } else {
+            return 0;
+        }
+
+        bestOfferId = otc.getBestOffer(wrappedSrc, wrappedDest);
+        (bestOfferSrcQty, , bestOfferDestQty,) = otc.getOffer(bestOfferId);
+
+        // make sure to take only first level of order book to avoid gas inflation.
+        if (actualSrcQty > bestOfferSrcQty) return 0;
+
+        rate = calcRateFromQty(bestOfferSrcQty, bestOfferDestQty, COMMON_DECIMALS, COMMON_DECIMALS);
```

Comment:
this will revert if qty is too big right?
It is fine we are sure the resolution is ok.